osu.Game/Overlays/News/Displays/ArticleListing.cs

```diff
@@ -20,26 +18,20 @@ namespace osu.Game.Overlays.News.Displays
     /// </summary>
     public class ArticleListing : CompositeDrawable
     {
-        public Action<APINewsSidebar> SidebarMetadataUpdated;
-
-        [Resolved]
-        private IAPIProvider api { get; set; }
+        public Action RequestMorePosts;
 
         private FillFlowContainer content;
         private ShowMoreButton showMore;
 
-        private GetNewsRequest request;
-        private Cursor lastCursor;
-
-        private readonly int? year;
+        private readonly GetNewsResponse initialResponse;
 
         /// <summary>
         /// Instantiate a listing for the specified year.
         /// </summary>
-        /// <param name="year">The year to load articles from. If null, will show the most recent articles.</param>
-        public ArticleListing(int? year = null)
+        /// <param name="initialResponse">Initial response to create articles from.</param>
+        public ArticleListing(GetNewsResponse initialResponse)
```

Comment:
This shouldn't be passed in like this. Or if it is, it should use a single implementation of the handling logic (right now you have it once in BDL and once in `AddPosts`).

Also, the whole response shouldn't be passed in; only the articles.

The show more button should probably be handled via a bindable. Or just a `bool` passed alongside the articles to keep things simple.